application.py

```diff
@@ -0,0 +1,207 @@
+import csv
+import datetime
+import re
+
+import matplotlib.pyplot as plt
+
+"""
+    Initialized categories list to choose correct option for add new spending.
+    todo:: add some more categories, create custom categories
+    todo:: (add them to other file and read to list to manipulate with them!)
+"""
+
+categoriesList = ["Food", "Clothes", "Entertainment", "Transportation", "Other"]
+
+
+def parseCategorieToNumber(category: str) -> int:
+    for i in range(len(categoriesList)):
+        if categoriesList[i] == category:
+            return i + 1
+    return -1
+
+
+class spending:
+    category: str
+    amount: float
+    date: str
+
+    def __init__(self, category: str, amount: float, date: str):
+        self.category = category
+        self.amount = amount
+        self.date = date
+
+    def toString(self) -> str:
+        return f"{self.category},{self.date},{self.amount}"
+
+class fileManipulation:
+    filename: str = "spending.csv"
+
+    """
+        Class to manipulate the file with the spending data.
+        Write spending object to csv file and read from csv file.
+    """
+
+    def writeToFile(self, spending: spending) -> None:
+        with open(self.filename, "a", newline="") as file:
+            writer = csv.writer(file)
+            writer.writerow([spending.category, spending.date, spending.amount])
+
+    def readFromFile(self) -> None:
+        with open(self.filename, "r") as file:
+            reader = csv.reader(file)
+            for row in reader:
+                print(row[0] + " " + row[1] + " " + row[2])
+   
+   
+    """
+    @input date: str in format year-month. e.g: 2021-01
+    @return list of spending objects 
+    """
+    def readFromFile_byDate(self, date: str) -> list or None:
+        result = re.match(r"\d{4}-\d{2}", date )
+        if result is None:
+            print("Invalid date format!")
+            return
+
+        list = []
+        with open(self.filename, "r") as file:
+            reader = csv.reader(file)
+            for row in reader:
+                if date in row[1]:
+                    list.append(spending(row[0], float(row[2]), row[1]))
+
+        return list
+
+    """
+    @input category: str in format year-month. e.g: 2021-01
+    @return list of spending objects
+    """
+    def readFromFile_byCategory(self, category: str) -> list[spending]:
+        list = []
+        with open(self.filename, "r") as file:
+            reader = csv.reader(file)
+            for row in reader:
+                if category in row[0]:
+                    list.append(spending(row[0], float(row[2]), row[1]))
+
+        return list
+
+
+"""
+    Simple interface to connect with user.
+    todo:: add some exception handlings in IO operations
+"""
+
+
+class menu:
+    def __init__(self) -> None:
+        self.fileManipulation = fileManipulation()
```

Comment:
lets pass such things as file manipulator as constructor parameter to have ability to replace it by another one in the future, if necessaary